ansible_collections/arista/avd/roles/eos_designs/python_modules/inband_management/avdstructuredconfig.py

```diff
@@ -179,32 +248,50 @@ def get_local_inband_mgmt_interface_cfg(self) -> dict:
                 "mtu": self.shared_utils.inband_mgmt_mtu,
                 "vrf": self.shared_utils.inband_mgmt_vrf,
                 "ip_address": self.shared_utils.inband_mgmt_ip,
+                "ipv6_enable": None if not self.shared_utils.configure_inband_mgmt_ipv6 else True,
+                "ipv6_address": self.shared_utils.inband_mgmt_ipv6_address,
                 "type": "inband_mgmt",
             }
         )
 
-    def get_parent_svi_cfg(self, vlan: int, subnet: str) -> dict:
-        network = ip_network(subnet, strict=False)
+    def get_parent_svi_cfg(self, vlan: int, subnet: str | None, ipv6_subnet: str | None) -> dict:
+        svidict = {
+            "name": f"Vlan{vlan}",
+            "description": self.shared_utils.inband_mgmt_description,
+            "shutdown": False,
+            "mtu": self.shared_utils.inband_mgmt_mtu,
+            "vrf": self.shared_utils.inband_mgmt_vrf,
+        }
+
+        network = False
+        v6_network = False
+
+        if subnet is not None:
+            network = ip_network(subnet, strict=False)
+            svidict["ip_attached_host_route_export"] = {"enabled": True, "distance": 19}
+
+        if ipv6_subnet is not None:
+            svidict["ipv6_enable"] = True
+            v6_network = ip_network(ipv6_subnet, strict=False)
+            svidict["ipv6_attached_host_route_export"] = {"enabled": True, "distance": 19}
 
         if self.shared_utils.mlag_role == "secondary":
-            ip = str(network[3])
+            if network:
+                ip = str(network[3])
+            if v6_network:
+                ipv6 = str(v6_network[3])
         else:
-            ip = str(network[2])
-        ip_address = f"{ip}/{network.prefixlen}"
-        gateway = str(network[1])
+            if network:
+                ip = str(network[2])
+            if v6_network:
+                ipv6 = str(v6_network[2])
 
-        return strip_empties_from_dict(
-            {
-                "name": f"Vlan{vlan}",
-                "description": self.shared_utils.inband_mgmt_description,
-                "shutdown": False,
-                "mtu": self.shared_utils.inband_mgmt_mtu,
-                "vrf": self.shared_utils.inband_mgmt_vrf,
-                "ip_address": ip_address,
-                "ip_virtual_router_addresses": [gateway],
-                "ip_attached_host_route_export": {
-                    "enabled": True,
-                    "distance": 19,
-                },
-            }
-        )
+        if network:
+            svidict["ip_address"] = f"{ip}/{network.prefixlen}"
+            svidict["ip_virtual_router_addresses"] = [str(network[1])]
+
+        if v6_network:
+            svidict["ipv6_address"] = f"{ipv6}/{v6_network.prefixlen}"
+            svidict["ipv6_virtual_router_addresses"] = [str(v6_network[1])]
```

Comment:
And remove the lines above checking for secondary and checking for subnets and so on and even remove the definition of netwrok 266 and v6_networks 267.

```suggestion
        if subnet is not None:
            network = ip_network(subnet, strict=False)
            ip = str(network[3]) if self.shared_utils.mlag_role == "secondary" else str(network[2])
            svidict["ip_attached_host_route_export"] = {"enabled": True, "distance": 19}
            svidict["ip_address"] = f"{ip}/{network.prefixlen}"
            svidict["ip_virtual_router_addresses"] = [str(network[1])]

        if ipv6_subnet is not None:
            v6_network = ip_network(ipv6_subnet, strict=False)
            ipv6 = str(v6_network[3]) if self.shared_utils.mlag_role == "secondary" else str(v6_network[2])
            svidict["ipv6_address"] = f"{ipv6}/{v6_network.prefixlen}"
            svidict["ipv6_enable"] = True
            svidict["ipv6_attached_host_route_export"] = {"enabled": True, "distance": 19}
            svidict["ipv6_virtual_router_addresses"] = [str(v6_network[1])]
```